agent/connect/ca/provider_vault.go

```diff
@@ -709,6 +722,66 @@ func (v *VaultProvider) Stop() {
 
 func (v *VaultProvider) PrimaryUsesIntermediate() {}
 
+// We use raw path here
+func (v *VaultProvider) mountNamespaced(namespace, path string, mountInfo *vaultapi.MountInput) error {
+	defer v.client.SetNamespace(v.baseNamespace)
+	if namespace != "" {
+		v.client.SetNamespace(namespace)
+	}
+
+	r := v.client.NewRequest("POST", fmt.Sprintf("/v1/sys/mounts/%s", path))
+	if err := r.SetJSONBody(mountInfo); err != nil {
```

Comment:
I see that we use raw `client.RawRequest` and sort-of manually interrogate Vault's HTTP API in lots of places in this provider. I'm guessing this is because we can no longer use `client.Sys` because of the need to inject the namespace into the path? Could we use `client.Logical` instead, though? (sorry in advance if I've missed what you were trying to say in your PR description)

Reply:
Caught up about this in-person (well, Zoom 😅) — the logical API client isn't suitable for this.